#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.1843534619556241 0.2632544700845438 -0.946948195544763 1.2735035387258107
  position 0.6313655238739713 2.5045048855282865 7.3747925354707595
}
TexturedBackground {
  texture "noon_cloudy_countryside"
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 1 1 7
  floorSize 3 3
}
Robot {
  translation 1 1 7.014980380003003
  rotation 0.8628029218401566 0.5053519313064264 0.013803752712705378 8.58822505737818e-15
  children [
    Camera {
      translation 0.01 0.01 0.1
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.1
          }
        }
      ]
    }
    DEF BODY Transform {
      translation 0 0 -0.015
      scale 7 7 7
      children [
        Group {
        }
        CadShape {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/bitcraze/crazyflie/protos/meshes/cf2_assembly.dae"
          ]
        }
      ]
    }
    DEF MULTIRANGER Transform {
      translation 0 0 0.02
      children [
        DistanceSensor {
          rotation 0 0 1 -1.57
          name "range_right"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 3.14
          name "range_back"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 1.57
          name "range_left"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          name "range_front"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
      ]
    }
    Camera {
      translation 0.03 0 0.01
      fieldOfView 0.87
      width 324
      height 324
    }
    GPS {
    }
    Gyro {
    }
    InertialUnit {
    }
    DEF M1 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 -0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m1_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation 0.031 -0.031 0.008
        children [
          DEF PROP_SHAPE Shape {
            appearance DEF PLASTIC PBRAppearance {
              baseColor 0 0 0
              metalness 0.3
            }
            geometry Mesh {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/bitcraze/crazyflie/protos/meshes/ccw_prop.dae"
              ]
            }
          }
        ]
      }
    }
    DEF M2 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 -0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m2_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 -0.031 0.008
        children [
          USE PROP_SHAPE
        ]
      }
    }
    DEF M3 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m3_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 0.031 0.008
        children [
          USE PROP_SHAPE
        ]
      }
    }
    DEF M4 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m4_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix DEF PROP Solid {
        translation 0.031 0.031 0.007
        children [
          USE PROP_SHAPE
        ]
      }
    }
  ]
  name "Crazyflie"
  model "Bitcraze's Crazyflie"
  boundingObject Cylinder {
    height 0.03
    radius 0.05
  }
  physics Physics {
    density -1
    mass 0.05
  }
  controller "drone"
  linearVelocity 1.1055719976262884e-17 -2.395959669107593e-18 -2.3468196285593958e-11
  angularVelocity -7.930164461608261e-16 9.540547003647312e-17 1.3671063617009837e-16
}
